src/os/bluestore/bluestore_tool.cc

```diff
@@ -113,6 +110,257 @@ BlueFS *open_bluefs(
       exit(EXIT_FAILURE);
     }
   }
+}
+
+void log_dump(
+  CephContext *cct,
+  const string& path,
+  const vector<string>& devs)
+{
+  validate_path(cct, path, true);
+  BlueFS *fs = new BlueFS(cct);
+  
+  add_devices(fs, cct, devs);
+  
+  int r = fs->open_super();
+  if (r < 0) {
+    cerr << "open_super failed" << ": "
+         << cpp_strerror(r) << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  BlueFS::FileRef log_file = new BlueFS::File;
+  bluefs_super_t super = fs->get_super();  
+  log_file->fnode = super.log_fnode;
+  cout << __func__ << " log_fnode " << super.log_fnode << std::endl;
+  
+  BlueFS::FileReader *log_reader = new BlueFS::FileReader(
```

Comment:
Can we try to reuse `BlueFS::_replay(bool noop)` instead of  duplicating it here? Otherwise it is going to be very fragile..